packages/react-server-cli/src/buildWebpackCompilers.js

```diff
@@ -0,0 +1,208 @@
+	let routesOutput = [];
+
+	const coreMiddleware = require.resolve("react-server-core-middleware");
+	const existingMiddleware = routes.middleware ? routes.middleware.map((middlewareRelativePath) => {
+		return `unwrapEs6Module(require("${path.relative(workingDirAbsolute, path.resolve(routesDir, middlewareRelativePath))}"))`
+	}) : [];
+	routesOutput.push(`
+var manifest = ${manifest ? JSON.stringify(manifest) : "undefined"};
+function unwrapEs6Module(module) { return module.__esModule ? module.default : module }
+var coreJsMiddleware = require('${coreMiddleware}').coreJsMiddleware;
+var coreCssMiddleware = require('${coreMiddleware}').coreCssMiddleware;
+module.exports = {
+	middleware:[
+		coreJsMiddleware(${JSON.stringify(staticUrl)}, manifest),
+		coreCssMiddleware(${JSON.stringify(staticUrl)}, manifest),
+		${existingMiddleware.join(",")}
+	],
+	routes:{`);
+
+	for (let routeName of Object.keys(routes.routes)) {
+		let route = routes.routes[routeName];
+
+		routesOutput.push(`
+		${routeName}: {`);
+		routesOutput.push(`
+			path: ${JSON.stringify(route.path)},`);
+		// if the route doesn't have a method, we'll assume it's "get". routr doesn't
+		// have a default (method is required), so we set it here.
+		routesOutput.push(`
+			method: "${route.method || "get"}",`);
+
+		let formats = normalizeRoutesPage(route.page);
+		routesOutput.push(`
+			page: {`);
+		for (let format of Object.keys(formats)) {
+			const formatModule = formats[format];
+			var relativePathToPage = path.relative(workingDirAbsolute, path.resolve(routesDir, formatModule));
+			routesOutput.push(`
+				${format}: function() {
+					return {
+						done: function(cb) {`);
+			if (isClient) {
+				routesOutput.push(`
+							require.ensure("${relativePathToPage}", function() {
+								cb(unwrapEs6Module(require("${relativePathToPage}")));
+							});`);
+			} else {
+				routesOutput.push(`
+							try {
+								cb(unwrapEs6Module(require("${relativePathToPage}")));
+							} catch (e) {
+								console.error('Failed to load page at "${relativePathToPage}"', e.stack);
+							}`);
+			}
+			routesOutput.push(`
+						}
+					};
+				},`);
+		}
+		routesOutput.push(`
+			},
+		},`);
+	}
+	routesOutput.push(`
+	}
+};`);
+
+	const routesContent = routesOutput.join("");
+	// make a unique file name so that when it is required, there are no collisions
+	// in the module loader between different invocations.
+	//const routesMD5 = crypto.createHash('md5').update(routesContent).digest("hex");
+	const routesFilePath = `${workingDirAbsolute}/routes_${isClient ? "client" : "server"}.js`;
+	fs.writeFileSync(routesFilePath, routesContent);
+
+	return routesFilePath;
+}
+
+// writes out a bootstrap file for the client which in turn includes the client
+// routes file. note that outputDir must be the same directory as the client routes
+// file, which must be named "routes_client".
+function writeClientBootstrapFile(outputFile, opts) {
```

Comment:
This file doesn't need to be actually written to disk. There are several possibilities to achieve the same effect. You could use the DefinePlugin to define the log levels. You could also use the imports-loader to inject build time variables.